### PR TITLE
ACS-3358 Support isShared field.

### PR DESCRIPTION
**Pull request by tpage-alfresco**

### CODE REVIEW

**Comment by tpage-alfresco** [comment]
This requires the release of https://github.com/Alfresco/alfresco-tas-restapi/pull/229 before the build can be run.

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java [hunk 1]
```diff
@@ -252,6 +253,20 @@ public void createRulesWithOneError()
...
+    public void createRuleAndIncludeFieldsInResponse()
+    {
+        RestRuleModel ruleModel = createRuleModel("ruleName");
+
+        RestRuleModel rule = restClient.authenticateUser(user).withCoreAPI().usingNode(ruleFolder).usingDefaultRuleSet()
+                                       .include("isShared")
+                                       .createSingleRule(ruleModel);
+
+        restClient.assertStatusCodeIs(CREATED);
+        rule.assertThat().field("isShared").isNotNull();
```
Comment (KikoUK) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:267]: Is `isShared` a mandatory/always returned in the response field? If yes, I think we can hint that is tested in the comment/description and maybe update one of the GET tests to include `isShared` assert

Comment (tpage-alfresco) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:267]: `isShared` is an optional field that needs requesting with `?include=isShared`

Comment (KikoUK) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:267]: Thanks. In that case I think we should update the description/comments of the existing tests you've updated with the` isShared` functionality to make it clear that it's been tested there as currently the comments only state that only basic functionality is tested.

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java [hunk 2]
```diff
@@ -83,7 +83,8 @@ public void createRule()
 
         restClient.assertStatusCodeIs(CREATED);
         rule.assertThat().field("id").isNotNull()
-            .assertThat().field("name").is("ruleName");
+            .assertThat().field("name").is("ruleName")
+            .assertThat().field("isShared").isNull();
```
Comment (KikoUK) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:87]: Please see above comment.

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/GetRulesTests.java [hunk 3]
```diff
@@ -102,8 +102,9 @@ public void getRulesList()
         rules.assertThat().entriesListCountIs(createdRules.size());
         IntStream.range(0, createdRules.size()).forEach(i ->
                 rules.getEntries().get(i).onModel()
-                         .assertThat().field("id").is(createdRules.get(i).getId())
-                         .assertThat().field("name").is(createdRules.get(i).getName()));
+                     .assertThat().field("id").is(createdRules.get(i).getId())
+                     .assertThat().field("name").is(createdRules.get(i).getName())
+                     .assertThat().field("isShared").isNull());
```
Comment (KikoUK) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/GetRulesTests.java:107]: Please see above comment.

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/GetRulesTests.java [hunk 4]
```diff
@@ -138,7 +154,8 @@ public void getSingleRule()
         restClient.assertStatusCodeIs(OK);
 
         rule.assertThat().field("id").is(createdRuleA.getId())
-            .assertThat().field("name").is(createdRuleA.getName());
+            .assertThat().field("name").is(createdRuleA.getName())
+            .assertThat().field("isShared").isNull();
```
Comment (KikoUK) [inline packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/GetRulesTests.java:158]: Please see above comment.

File: packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java (hunk 1 above)
Comment (tpage-alfresco) [reply packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/rules/CreateRulesTests.java:271]: Good suggestion - done in 64b3e61e1cdf7fae338d8158ec106278fd242bf5

**Review by KikoUK — APPROVED** [review]

**Comment by tpage-alfresco** [comment]
The only failure was:

> Failed to execute goal com.srcclr:srcclr-maven-plugin:3.1.21:scan

I assume there's an issue with Veracode, so merging anyway.

**Pull request merged**